app/controllers/checkout_forms/billing_address_form.rb

```diff
@@ -0,0 +1,24 @@
+class BillingAddressForm < AddressForm
+  def save
+    if valid?
+      persist!
```

Comment:
won't persist! return true if it succeeds? So, maybe you don't need to return true in the next line?

```
def save
  return false if invalid?
  persist!
end
```